recipe/meta.yaml

```diff
@@ -10,7 +10,7 @@ source:
     sha256: dceb15fef556a2775121bcc695561df4ff0e09248e0ce3a2d58f5244a9f61421
 
 build:
-    number: 0
+    number: 1
```

Comment:
I see.  OK then.  I think we really need some more docs about what happens when. I didn't know that `upload_or_check_non_existence` existed and I can't seem to find it in the usual spots.

We also should capture in docs what you just wrote about when to update the build number. I am pretty sure I read somewhere that you change it whenever you don't change the version number and reset to zero if you do change the version number.  What you just wrote is a little more subtle than that :)